ClassOrdering reports a message describing the misorder

hbmartin opened the pull request:
* Currently the `ClassOrdering` rule reports its own description as a message. This PR modifies the message to report the specific entities identified as out of order.
* It also slightly refactors the class organization, but requires no test changes, only test changes exist to check message
* Additional tests (for test coverage sake) describe existing behavior
* Future PR could improve this by reporting all misordered pairs, not simply the first one: https://github.com/detekt/detekt/issues/3141
* Follow up to: https://github.com/detekt/detekt/pull/3088

BraisGabin reviewed (commented):
Great first PR!

Could you add some asserts over the message in the tests? The code that build the message is kind of complex. 

And, could you add the missing parts that you found from this rule in a new issue so we can track them? If you want to do those PRs we can assign that issue to you.

hbmartin commented:
@BraisGabin thanks!
Added tests for message: https://github.com/detekt/detekt/pull/3138/commits/3dc635511b7ca02575662d768ecb90148490c336
Added an issue for tracking the list of misorders issue: https://github.com/detekt/detekt/issues/3141

BraisGabin commented on detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ClassOrderingSpec.kt:
```diff
@@ -74,7 +74,10 @@ class ClassOrderingSpec : Spek({
+            val findings = subject.compileAndLint(code)
+            assertThat(findings).hasSize(1)
+            assertThat(findings[0].message).isEqualTo("OutOfOrder (secondary constructor) " +
+                "should not come before null (class initializer)")
```
`null` is kind of strange here, right?

hbmartin replied:
Yea, so is `Companion (companion)` - fixed in 511e8c9680f57718a3b0fee2350bb539aa82ceda

BraisGabin reviewed (approved):
Nice job!!

hbmartin commented:
@BraisGabin thanks! how does this get merged?

schalkms reviewed (approved):
Thanks! @hbmartin 🙂
